packages/langium/src/workspace/documents.ts

```diff
@@ -327,6 +404,17 @@ export class DefaultLangiumDocuments implements LangiumDocuments {
     }
 
     invalidateDocument(uri: URI): void {
+        const uriString = uri.toString();
+        const langiumDoc = this.documentMap.get(uriString);
+        if (langiumDoc) {
+            langiumDoc.state = DocumentState.Changed;
+            langiumDoc.references = [];
+            langiumDoc.precomputedScopes = undefined;
+            langiumDoc.diagnostics = [];
+        }
+    }
+
+    deleteDocument(uri: URI): void {
```

Comment:
Thought: shall we keep the document state intact in this new method? Rationale: the method name does not imply that the content has changed.